### PR TITLE
[skip circleci] Rename hlint test data files and add regression tests

### PR DESCRIPTION
**Pull request by jneira**
* Change test module cases from ApplyRefactN to more descriptive names to help in its discovery
* Add regression tests for #2042 #2280 and a rperoduction test for #2290
* Correct `waitForProgressDone` hls-test-utils implementation: it was a exact duplicate of `waitForAllProgressDone` so it is replaced with the definition from the same function in the ghcide test suite

<a href="https://gitpod.io/#https://github.com/haskell/haskell-language-server/pull/2321"></a>

### CODE REVIEW

**Comment by jneira** [comment]
Tests are green after the usual reruns, so I think it could be reviewed although most changes are naming related

File: plugins/hls-hlint-plugin/test/Main.hs [hunk 1]
```diff
@@ -285,3 +316,9 @@ hlintConfigWithFlags flags =
   where
     unObject (Object obj) = obj
     unObject _            = undefined
+
+knownBrokenForHlintOnGhcLib :: String -> TestTree -> TestTree
```
Comment (michaelpj) [inline plugins/hls-hlint-plugin/test/Main.hs:320]: It's not immediately obvious to me what these mean or why it's these versions. Haddock?

**Review by michaelpj — APPROVED** [review]
> Correct waitForProgressDone hls-test-utils implementation: it was a exact duplicate of waitForAllProgressDone so it is replaced with the definition from the same function in the ghcide test suite

Can we share this? Perhaps we need to upstream some more test helpers into `lsp-test`?

File: plugins/hls-hlint-plugin/test/Main.hs (hunk 1 above)
Comment (jneira) [reply plugins/hls-hlint-plugin/test/Main.hs:320]: Hmm it is a somewhat long history, will try to add a concise comment although the entire thing should be explained in the plugin README or something

Comment (jneira) [reply plugins/hls-hlint-plugin/test/Main.hs:327]: Not sure if i explained it clearly :thinking:

**Comment by jneira** [comment]
Err why mergify merged the pr without waiting for required checks?, argh

**Comment by jneira** [comment]
I think the post_job's are not shown until they are finished, so if test postjob has not finished but benchmark or nix one does, you can merge the pr, will try to fix giving them unique names

**Pull request merged**